README.md

```diff
@@ -251,6 +262,77 @@ The shared storage worklet invocation methods (`addModule`, `createWorklet`, and
 *  See example usage below.
 
+### Locking for Modifier Methods
+
+All modifier methods (`set`, `append`, `delete`, `clear`), whether invoked from JavaScript or from response headers, accept a `withLockOnResource: <resource>` option. This option instructs the method to acquire a lock on the designated resource before executing.
+
+The locks requested this way are partitioned by the shared storage data origin, and are independent of any locks obtained via `navigator.locks.request` in a Window or Worker context. Note that they share the same scope with the locks obtained via `navigator.locks.request` in the SharedStorageWorklet context.
+
+Unlike `navigator.locks.request`, which offers various configuration options, the locks requested this way always use the default settings:
+*  `mode: "exclusive"`: The lock is never shared with other locks.
+*  `steal: false`: The lock will not preempt other locks.
+*  `ifAvailable: false`: If the lock is currently held by others, keep waiting and don't skip.
+
+#### Example: Naive Integrity Check
+
+This naive example uses a lock to ensure that the integrity check within the worklet runs atomically, preventing interference from other operations, such as `sharedStorage.set()` outside the worklet.
+
+Window context:
+
+```js
+try {
+  sharedStorage.set('key', generateRandom(), {withLockOnResource: 'naive-integrity-check-lock'});
+
+  await window.sharedStorage.worklet.addModule('naive-integrity-check-script.js');
+  await window.sharedStorage.worklet.run('naive-integrity-check');
+} catch (error) {
+  // Handle error.
+}
+```
+
+In the worklet script (`naive-integrity-check-script.js`):
+
+```js
+class NaiveIntegrityCheckOperation {
+  async run(data) {
+    await navigator.locks.request("naive-integrity-check-lock", async (lock) => {
+      if (await sharedStorage.get('key') != await sharedStorage.get('key')) {
+        throw Error('Invalid state!');
+      }
+    });
+  }
+}
+register('naive-integrity-check', NaiveIntegrityCheckOperation);
+```
+
+#### Caveat 1: Unexpected ordering
+
+Modifier methods may block due to the lock, so may not execute in the order they appear in the code.
+
+```js
+// Resolve immediately. Internally, this may block to wait for the lock to be granted.
+sharedStorage.set('key0', 'value1', { withLockOnResource: 'resource0' });
+
+// Resolve immediately. Internally, this will also execute immediately.
+sharedStorage.set('key0', 'value2');
+```
+
+Developers should be mindful of this potential ordering issue.
+
+#### Caveat 2: No atomicity for multiple writes
+
+The `withLockOnResource` option provides locking for individual methods but doesn't guarantee atomicity across multiple methods.
+
+Developers should be mindful of this limitation. To get around, consider one of these options:
+1.  Move the modifier methods inside the worklet. This allows a single lock to protect them.
```

Comment:
Needs a bit more description, e.g., that you need to send the data via the data parameter to the worklet.